concent_api/conductor/models.py

```diff
@@ -0,0 +1,90 @@
+
+    # Relative path of the .zip file that contains the rendering result received from the provider.
+    result_package_path = CharField(max_length=255, unique=True)
+
+    def clean(self):
+        super().clean()
+
+        # source_package_path cannot be used as result_package_path in any model instance.
+        if VerificationRequest.objects.filter(result_package_path=self.source_package_path).exists():
+            raise ValidationError({
+                'source_package_path': 'source_package_path cannot be used as result_package_path in any other VerificationRequest.'
+            })
+
+        # result_package_path cannot be used as source_package_path in any model instance.
+        if VerificationRequest.objects.filter(source_package_path=self.result_package_path).exists():
+            raise ValidationError({
+                'result_package_path': 'result_package_path cannot be used as source_package_path in any other VerificationRequest.'
+            })
+
+
+class BlenderSubtaskDefinition(Model):
+    """
+    For each VerificationRequest there must be exactly one BlenderSubtaskDefinition in the database.
+    """
+
+    class OutputFormat(ChoiceEnum):
+        JPG = 'jpg'
+        JPEG = 'jpeg'
+        BMP = 'bmp'
+        SGI = 'sgi'
+        RGB = 'rgb'
+        BW = 'bw'
+        PNG = 'PNG'
+        JP2 = 'jp2'
+        JP2C = 'jp2c'
+        TGA = 'tga'
+        CIN = 'cin'
+        DPX = 'dpx'
+        HDR = 'hdr'
+        TIF = 'tif'
+        TIFF = 'tiff'
+
+    # Foreign key to VerificationRequest. Can't be NULL and must be unique.
+    verification_request = OneToOneField(VerificationRequest, unique=True, related_name='blender_subtask_definition')
+
+    # Type of the output image to be produced by Blender. This determines the file extensions.
+    # Only formats supported by Blender should be allowed here.
+    # This value is passed to Blender using the -F command-line option.
+    output_format = CharField(max_length=32, choices=OutputFormat.choices())
+
+    # Relative path to the .blend file inside the source package.
+    scene_file = CharField(max_length=MESSAGE_PATH_LENGTH)
+
+    # Indicates when Conductor has received the request.
+    created_at = DateTimeField(default=timezone.now)
+
+
+class UploadReport(Model):
+    """
+    Existence of this object indicates that a file has been uploaded to nginx-storage
+    and nginx notified Conductor about this fact.
+    """
+
+    # Relative path of the file. Relative to the same directory that paths listed in FileTransferTokens are relative to.
+    path = CharField(max_length=MESSAGE_PATH_LENGTH)
+
+    # Foreign key to VerificationRequest. Can be NULL if there's no corresponding request.
+    verification_request = ForeignKey(VerificationRequest, related_name = 'upload_reports', blank=True, null=True)
+
+    # Indicates when conductor has been notified about the upload.
+    created_at      = DateTimeField(default = timezone.now)
```

Comment:
There seams to be a lot of "space" issue i this whole PR. Which style we should use anyway? With or without spaces in function call?

Reply:
@Jakub89 All should be updated by now.